Core.ApplicationServices/GDPR/Write/DataProcessingRegistrationWriteService.cs

```diff
@@ -330,6 +333,23 @@ private Maybe<OperationError> UpdateBasisForTransfer(DataProcessingRegistration
                 .MatchFailure();
         }
 
+        private Result<DataProcessingRegistration, OperationError> UpdateMainContract(DataProcessingRegistration dpr, Guid? contractId)
+        {
+            if (!contractId.HasValue)
+            {
```

Comment:
funny how the "reset" actually exists when you use another one in the app service